### PR TITLE
Remove HasNSeriesSKU

### PR DESCRIPTION
**Pull request by bowang-666**
HasNSeriesSKU used case sensitive matching, which cause driver configuration to be skipped if sku name doesn't match "Stardard_N"

### CODE REVIEW

File: pkg/agent/baker.go [hunk 1]
```diff
@@ -438,7 +438,7 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 		"AnyAgentIsLinux": func() bool {
 			return cs.Properties.AnyAgentIsLinux()
 		},
-		"IsNSeriesSKU": func(profile *datamodel.AgentPoolProfile) bool {
+		"IsNSeriesSKU": func() bool {
 			return config.EnableNvidia
 		},
```
Comment (gtxistxgao) [inline pkg/agent/baker.go:443]: the function should return if this SKU is N series. The function is doing another thing. will it possible that EnableNvidia != NSeriesSKU?

Comment (bowang-666) [reply pkg/agent/baker.go:443]: No EnableNvidia is determined in RP https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp?path=%2Fresourceprovider%2Fserver%2Fmicrosoft.com%2Fcommon%2Fagentbaker%2Fagentbakerhelper.go&version=GBmaster&line=88&lineEnd=88&lineStartColumn=3&lineEndColumn=80&lineStyle=plain&_a=contents
where it's compared case insensitive

**Review by adilliadil — APPROVED** [review]
lgtm

File: pkg/agent/baker.go (hunk 1 above)
Comment (gtxistxgao) [reply pkg/agent/baker.go:443]: yea my concern is a nit that the naming of the function does not exactly matching what it says. functionality looks good to me.

**Pull request merged**